.github/workflows/e2e-nightly.yml

```
@@ -0,0 +1,45 @@
+# Runs randomly generated E2E testnets nightly.
+name: e2e-nightly
+on:
+  pull_request: # FIXME temporary, to test the action
+  workflow_dispatch: # allow running workflow manually
+  schedule:
+    - cron: '0 2 * * *'
+
+jobs:
+  test:
+    # Run parallel jobs for the listed testnet groups (must match the
+    # ./build/generator -g flag)
+    strategy:
+      fail-fast: false
+      matrix:
+        group: ['00', '01', '02', '03']
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build
+        working-directory: test/e2e
+        # Run make jobs in parallel, since we can't run steps in parallel.
+        run: make -j2 docker generator runner
+
+      - name: Generate testnets
+        working-directory: test/e2e
+        run: ./build/generator -g 4 -d networks/nightly
+
+      - name: Run testnets in group ${{ matrix.group }}
+        working-directory: test/e2e
+        run: sudo ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Notify Slack on failure
+        uses: rtCamp/action-slack-notify@e9db0ef
+        #if: ${{ failure() }}
+        env:
+          SLACK_WEBHOOK: # FIXME Set a webhook here
```

```suggestion
          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
```

you can do this for now if you would like to merge and when the secret becomes available it will pick it up

---

I'll wait for it to get set up, so that I can test it manually.